modules/clientBinaryManager.js

```diff
@@ -180,7 +180,9 @@ class Manager extends EventEmitter {
         .then((localConfig) => {
             if (!localConfig) {
                 log.info('No config for the ClientBinaryManager could be loaded, using local clientBinaries.json.');
-                localConfig = require('../clientBinaries.json');
+
+                const localConfigPath = path.join(Settings.userDataPath, 'clientBinaries.json');
```

Comment:
As far as i understand this block:
```js
// FETCH
 return got(BINARY_URL, {
             timeout: 3000,		             timeout: 3000,
             json: true,		             json: true,
         })		         })
// GET RESULT OR ERROR
         .then((res) => {		         .then((res) => {
             if (!res || _.isEmpty(res.body)) {		             if (!res || _.isEmpty(res.body)) {
                 throw new Error('Invalid fetch result');		                 throw new Error('Invalid fetch result');
             } else {		             } else {
                 return res.body;		                 return res.body;
             }		             }
         })		         })
// CATCH ERRORS from fetching, on or offline
         .catch((err) => {		         .catch((err) => {
             log.warn('Error fetching client binaries config from repo', err);		             log.warn('Error fetching client binaries config from repo', err);
         })		         })
// CONTINUE here, even if error was above thrown.
         .then((latestConfig) => {		         .then((latestConfig) => {
```

Please test this that a offline state will certainly not go into this then block after the catch and the duplication of the code is really necessary